src/Microsoft.CodeQuality.Analyzers/UnitTests/ApiDesignGuidelines/DoNotHideBaseClassMethodsTests.cs

```diff
@@ -0,0 +1,455 @@
+
+            VerifyBasic(@"
+MustInherit Class Base
+    Public MustOverride Sub Method(input As String)
+    End Sub
+End Class
+
+Class Derived
+    Inherits Base
+    
+    Public Overrides Sub Method(input As Object)
+    End Sub
+End Class",
+                TestValidationMode.AllowCompileErrors);
+        }
+
+        [Fact]
+        public void CA1061_DerivedMethodHasLessDerivedParameter_DerivedMethodPrivate_Diagnostic()
+        {
+            VerifyCSharp(@"
+class Base
+{
+    public void Method(string input)
+    {
+    }
+}
+
+class Derived : Base
+{
+    private void Method(object input)
+    {
+    }
+}",
+                GetCA1061CSharpResultAt(11, 18, "Derived.Method(object)", "Base.Method(string)"));
+
+            VerifyBasic(@"
+Class Base
+    Public Sub Method(input As String)
+    End Sub
+End Class
+
+Class Derived
+    Inherits Base
+
+    Public Sub Method(input As Object)
+    End Sub
+End Class
+",
+                GetCA1061BasicResultAt(10, 16, "Public Sub Method(input As Object)", "Public Sub Method(input As String)"));
+        }
+
+        [Fact]
+        public void CA1061_DerivedMethodHasLessDerivedParameter_BaseMethodPrivate_NoDiagnostic()
+        {
+            // Note: This behavior differs from FxCop's CA1061
+            VerifyCSharp(@"
+class Base
+{
+    private void Method(string input)
+    {
+    }
+}
+
+class Derived : Base
+{
+    public void Method(object input)
+    {
+    }
+}");
+
+            VerifyBasic(@"
+Class Base
+    Private Sub Method(input As String)
+    End Sub
+End Class
+
+Class Derived
+    Inherits Base
+
+    Public Sub Method(input As Object)
+    End Sub
+End Class
+");
+        }
+
+        [Fact]
+        public void CA1061_DerivedMethodHasLessDerivedParameter_ArityMismatch_NoDiagnostic()
+        {
+            VerifyCSharp(@"
+class Base
+{
+    public void Method(string input, string input2)
+    {
+    }
+}
+
+class Derived : Base
+{
+    public void Method(object input)
+    {
+    }
+}");
+
+            VerifyBasic(@"
+Class Base
+    Private Sub Method(input As String, input2 As String)
+    End Sub
+End Class
+
+Class Derived
+    Inherits Base
+
+    Public Sub Method(input As Object)
+    End Sub
+End Class
+");
+        }
+
+        [Fact]
+        public void CA1061_DerivedMethodHasLessDerivedParameter_ReturnTypeMismatch_NoDiagnostic()
+        {
+            VerifyCSharp(@"
+class Base
+{
+    public void Method(string input)
+    {
+    }
+}
+
+class Derived : Base
+{
+    public int Method(object input)
+    {
+        return 0;
+    }
+}");
+
+            VerifyBasic(@"
+Class Base
+    Private Sub Method(input As String)
+    End Sub
+End Class
+
+Class Derived
+    Inherits Base
+
+    Public Function Method(input As Object) As Integer
+        Method = 0
+    End Function
+End Class
+");
+        }
+
+        [Fact]
+        public void CA1061_DerivedMethodHasLessDerivedParameter_ParameterTypeMismatch_NoDiagnostic()
```

Comment:
Can you also add a test with parameter types reversed: `Base.Method(int input, string input2)` and `Derived.Method(char input, object input2)`